File: HackerRank/1- 30 Days of Code/Day 28 RegEx, Patterns, and Intro to Databases/answer.js
"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", function (inputStdin) {
  inputString += inputStdin;
});

process.stdin.on("end", function () {
  inputString = inputString.split("\n");

  main();
});

function readLine() {
  return inputString[currentLine++];
}

function main() {
  let N = parseInt(readLine());
  let pattern = new RegExp("@gmail.com");
  let names = [];
  for (let i = 0; i < N; i++) {
    let firstName_temp = readLine().split(" ");
    let firstName = firstName_temp[0];
    let emailID = firstName_temp[1];
    if (pattern.test(emailID)) {
      names.push(firstName);
    }
  }
  names.sort().forEach(function (name) {
    console.log(name);
  });
}

File: HackerRank/1- 30 Days of Code/Day 28 RegEx, Patterns, and Intro to Databases/answer.py
#!/bin/python3

import math
import os
import random
import re
import sys


if __name__ == '__main__':
    N = int(input().strip())

    pattern = r"@gmail\.com$"

    regex = re.compile(pattern)

    first_name_list = []

    for N_itr in range(N):
        first_multiple_input = input().rstrip().split()

        firstName = first_multiple_input[0]

        emailID = first_multiple_input[1]

        if regex.search(emailID):
            first_name_list.append(firstName)

    first_name_list.sort()

    for name in first_name_list:
        print(name)
